src/libsyncengine/reconciliation/platform_inconsistency_checker/platforminconsistencycheckerworker.cpp

```diff
@@ -104,66 +120,109 @@ ExitCode PlatformInconsistencyCheckerWorker::checkTree(std::shared_ptr<Node> rem
     return ExitCode::Ok;
 }
 
-void PlatformInconsistencyCheckerWorker::blacklistNode(const std::shared_ptr<Node> remoteNode, const SyncPath &relativePath,
+ExitCode PlatformInconsistencyCheckerWorker::checkLocalTree(std::shared_ptr<Node> localNode, const SyncPath &parentPath) {
+    if (localNode->hasChangeEvent(OperationType::Delete)) {
+        return ExitCode::Ok;
+    }
+
+    if (localNode->hasChangeEvent(OperationType::Create) || localNode->hasChangeEvent(OperationType::Move)) {
+        if (PlatformInconsistencyCheckerUtility::instance()->checkNameSize(localNode->name())) {
+            blacklistNode(localNode, InconsistencyType::NameLength);
+            return ExitCode::Ok;
+        }
+    }
+
+    auto childeIt = localNode->children().begin();
+    for (; childeIt != localNode->children().end(); childeIt++) {
+        if (stopAsked()) {
+            return ExitCode::Ok;
+        }
+
+        while (pauseAsked() || isPaused()) {
+            if (!isPaused()) {
+                setPauseDone();
+            }
+
+            Utility::msleep(LOOP_PAUSE_SLEEP_PERIOD);
+        }
+
+        ExitCode exitCode = checkLocalTree(childeIt->second, parentPath / localNode->name());
+        if (exitCode != ExitCode::Ok) {
+            return exitCode;
+        }
+    }
+    return ExitCode::Ok;
+}
+
+void PlatformInconsistencyCheckerWorker::blacklistNode(const std::shared_ptr<Node> node,
                                                        const InconsistencyType inconsistencyType) {
     // Local node needs to be excluded before call to blacklistTemporarily because
     // we need the DB entry to retrieve the corresponding node
     NodeIdPair nodeIDs;
+    const std::shared_ptr<Node> localNode = node->side() == ReplicaSide::Local ? node : correspondingNodeDirect(node);
+    const std::shared_ptr<Node> remoteNode = node->side() == ReplicaSide::Remote ? node : correspondingNodeDirect(node);
 
-    if (const auto localNode = correspondingNodeDirect(remoteNode); localNode) {
-        // Also exclude local node by adding "conflict" suffix
+    if (localNode) {
         const SyncPath absoluteLocalPath = _syncPal->localPath() / localNode->getPath();
         LOGW_SYNCPAL_INFO(_logger,
-                          L"Excluding also local item with " << Utility::formatSyncPath(absoluteLocalPath).c_str() << L".");
-        PlatformInconsistencyCheckerUtility::renameLocalFile(absoluteLocalPath,
-                                                             PlatformInconsistencyCheckerUtility::SuffixTypeConflict);
+                          L"Excluding local item with " << Utility::formatSyncPath(absoluteLocalPath).c_str() << L".");
+        PlatformInconsistencyCheckerUtility::renameLocalFile(
+                absoluteLocalPath, node->side() == ReplicaSide::Remote
```

Comment:
Create a small helper function. Naming it will already help understand the expected outcome.

Reply:
Could you please provide more insight about what you are thinking about. I do not see what the small helper functions should do here